### DISCUSSION TITLE
Edit & Commit to protecting branch should be warned if branch protection enabled

### DISCUSSION THREAD

**Discussion by kyontan** [community/community#36522] [Other Feature Feedback, Questions, & Ideas]
Hi folks,

I have enabled the branch protection: "Require a pull request before merging" to prevent unintentional push to the protected branch.

Though there is need for by-passing on emergency situation, we didn't enable "Do not allow bypassing the above settings" so that we can merge the Pull Request without any approves / checks if we surely understand it.

<img width="648" alt="image" src="https://user-images.githubusercontent.com/802339/196369962-2a502a82-a93e-4bb5-903d-235187644c7d.png">

Though the settings above enabled, we can directly push the commit to protected branch without any checks / warnings.
Recently it caused a small issue in our organization.

<img width="780" alt="image" src="https://user-images.githubusercontent.com/802339/196370159-81afa945-5f03-4a2a-9b1e-86c2ee56e521.png">

I'd like to have a warnings (like some attentions that is already at the Pull Request page) in editing page.

Thanks!

**Comment by ossanna16** [comment]
Hey there @kyontan and thank you for asking a good question! I'm glad you found our community 🙂
